### PR TITLE
131 create a client system to send a newly created entity

### PR DESCRIPTION
**Pull request by LucasTesnier**
Implement and link all the network functionnality.

### CODE REVIEW

File: Client/ClientRoom.cpp [hunk 1]
```diff
@@ -42,26 +46,54 @@ ClientRoom::ClientRoom(std::string address, unsigned short port, std::string ser
...
 
-void ClientRoom::startLobbyLoop(void)
-{
-    CommunicatorMessage connexionResponse;
-
-    std::size_t entityId = createNewObstacle(*(_worldInstance.get()), 5, 60, 5);
+struct Temp : public System {
+    void run(World &world)
+    {
+        (void)world;
```
Comment (31Nathan) [inline Client/ClientRoom.cpp:52]: Why ? Add a comment if this will be implemented later

File: Server/Room.cpp [hunk 2]
```diff
@@ -42,49 +52,108 @@ Room::Room(unsigned short id, Client networkInformations)
...
     _worldInstance.get()->setTransisthorBridge(_communicatorInstance.get()->getTransisthorBridge());
     _state = RoomState::UNDEFINED;
+    _remainingPlaces = 4;
 }
 
 struct Temp : public System {
-    void run(World &world)
-    {
-        std::vector<std::shared_ptr<Entity>> joined = world.joinEntities<Position>();
+    void run(World &world) { (void)world; }
```
Comment (31Nathan) [inline Server/Room.cpp:59]: Same, maybe I didn't understand something

File: Server/Room.cpp [hunk 3]
```diff
@@ -42,49 +52,108 @@ Room::Room(unsigned short id, Client networkInformations)
...
+            _worldInstance.get()->getTransisthorBridge()->transitEcsDataToNetworkDataEntityObstacle(
+                entityPtr->getComponent<Networkable>().id, pos.x, pos.y, entityPtr->getComponent<Damage>().damagePoint,
+                "", {connexionDemand.message.clientInfo.getId()});
+        }
+        if (entityPtr->contains<Player>()) {
+            Position &pos = entityPtr->getComponent<Position>();
+            Velocity &vel = entityPtr->getComponent<Velocity>();
+            Size &size = entityPtr->getComponent<Size>();
+
+            // if (tempId != entityPtr->getId()) {
```
Comment (31Nathan) [inline Server/Room.cpp:141]: Remove comment

File: Server/Room.cpp [hunk 4]
```diff
@@ -42,49 +52,108 @@ Room::Room(unsigned short id, Client networkInformations)
...
+    _worldInstance.get()->getEntity(tempId).addComponent<NetworkClient>(connexionDemand.message.clientInfo.getId());
+    std::vector<std::shared_ptr<Entity>> joined = _worldInstance.get()->joinEntities<Networkable>();
+
+    for (std::shared_ptr<Entity> entityPtr : joined) {
+        if (entityPtr->contains<AlliedProjectile>()) {
+            _worldInstance.get()->getTransisthorBridge()->transitEcsDataToNetworkDataEntityAlliedProjectile(
+                entityPtr->getComponent<Networkable>().id, entityPtr->getComponent<AlliedProjectile>().parentNetworkId, "",
+                {connexionDemand.message.clientInfo.getId()});
+        }
+        if (entityPtr->contains<Enemy>()) {
```
Comment (31Nathan) [inline Server/Room.cpp:113]: In part 2, you may think about splitting this function for each if

File: libs/Transisthor/Transisthor.cpp [hunk 5]
```diff
@@ -355,10 +361,45 @@ void Transisthor::entityConvertAlliedProjectileType(unsigned short id, void *byt
...
-    (void)id;
-    /// SEND THE NEW ENTITY TO ECS, WILL BE ADDED WHEN TRANSISTHOR WILL BE FULLY IMPLEMENTED
+    std::vector<std::shared_ptr<Entity>> networkables = _ecsWorld.joinEntities<Networkable>();
+
+    auto findShooter = [allyId](std::vector<std::shared_ptr<Entity>> networkables) {
+        for (std::shared_ptr<Entity> ptr : networkables) {
+            if (ptr->getComponent<Networkable>().id == allyId)
+                return ptr;
+        }
+        throw NetworkError("The entity was not find");
```
Comment (31Nathan) [inline libs/Transisthor/Transisthor.cpp:371]: Was not found

File: libs/Transisthor/Transisthor.cpp [hunk 6]
```diff
@@ -355,10 +361,45 @@ void Transisthor::entityConvertAlliedProjectileType(unsigned short id, void *byt
...
+    };
+    std::shared_ptr<Entity> shooter;
+    try {
+        shooter = findShooter(networkables);
+    } catch (const NetworkError &e) {
+        return;
```
Comment (31Nathan) [inline libs/Transisthor/Transisthor.cpp:377]: Add a logger message

**Review by 31Nathan — CHANGES_REQUESTED** [review]
Some minors changes, both of you are remarquables workers !

File: Server/Room.cpp (hunk 4 above)
Comment (LucasTesnier) [reply Server/Room.cpp:114]: :eyes:

File: libs/Transisthor/Transisthor.cpp [hunk 7]
```diff
@@ -355,10 +361,46 @@ void Transisthor::entityConvertAlliedProjectileType(unsigned short id, void *byt
...
-    (void)id;
-    /// SEND THE NEW ENTITY TO ECS, WILL BE ADDED WHEN TRANSISTHOR WILL BE FULLY IMPLEMENTED
+    std::vector<std::shared_ptr<Entity>> networkables = _ecsWorld.joinEntities<Networkable>();
+
+    auto findShooter = [allyId](std::vector<std::shared_ptr<Entity>> networkables) {
+        for (std::shared_ptr<Entity> ptr : networkables) {
+            if (ptr->getComponent<Networkable>().id == allyId)
+                return ptr;
+        }
+        throw NetworkError("The entity was not fouund");
```
Comment (31Nathan) [inline libs/Transisthor/Transisthor.cpp:371]: FOUUUUUUUUUUUUUUUUUUUUUUUUUUUUNDDDDDDDD

File: libs/Transisthor/Transisthor.cpp [hunk 8]
```diff
@@ -355,10 +361,46 @@ void Transisthor::entityConvertAlliedProjectileType(unsigned short id, void *byt
...
+    };
+    std::shared_ptr<Entity> shooter;
+    try {
+        shooter = findShooter(networkables);
+    } catch (const NetworkError &e) {
+        std::cerr << "Parent entities cannot be founded." << std::endl;
```
Comment (31Nathan) [inline libs/Transisthor/Transisthor.cpp:377]: Just found

**Review by 31Nathan — CHANGES_REQUESTED** [review]
Minor syntax errors

**Review by TomDUVAL-MAHE — APPROVED** [review]
NTR

**Review by TomDUVAL-MAHE — DISMISSED** [review]

File: libs/Transisthor/Transisthor.cpp (hunk 7 above)
Comment (31Nathan) [inline libs/Transisthor/Transisthor.cpp:371]: U seriouus ?

**Review by 31Nathan — CHANGES_REQUESTED** [review]
...

**Review by 31Nathan — APPROVED** [review]
GG BOYS ❤️

**Review by osvegn — APPROVED** [review]

**Review by ZQUEMA — APPROVED** [review]

**Pull request merged**